addons/website_slides/models/slide_slide.py

```diff
@@ -877,16 +897,48 @@ def _action_set_completed(self, target_partner):
             'vote': 0,
             'completed': True} for new_slide in new_slides])
 
+        self.env['slide.channel'].invalidate_cache(['completed', 'completion'], ids=self.channel_id.ids)
+
+        return True
+
+    def action_set_uncompleted(self):
+        if any(not slide.channel_id.is_member for slide in self):
+            raise UserError(_('You cannot mark a slide as uncompleted if you are not among its members.'))
+
+        if any(not slide.can_be_undone for slide in self):
+            raise UserError(_('Those slides can not be undone.'))
+
+        target_partner = self.env.user.partner_id
+
+        # Remove the Karma point gained
+        self._action_set_quiz_done(completed=False)
```

Comment:
This line is strange.
Not called in "_action_set_completed" so not sure why it's called here.

Reply:
Does that mean we could gain twice the karma then? Once when we call the route to submit the answers and once when we call the route to mark the slide as completed?

Reply:
Well "_action_set_quiz_done" checks the completion status of the slide and does a "continue".

Since it is now only called priori to setting the completion status to True/False, I think it should cover all use cases.